jax/_src/scipy/stats/_core.py

```diff
@@ -81,3 +81,64 @@ def _mode_helper(x: jnp.ndarray) -> Tuple[jnp.ndarray, jnp.ndarray]:
   x = x.reshape(x.shape[0], prod(x.shape[1:]))
   vals, counts = vmap(_mode_helper, in_axes=1)(x)
   return ModeResult(vals.reshape(output_shape), counts.reshape(output_shape))
+
+def invert_permutation(i: Array) -> Array:
+  """Helper function that inverts a permutation array."""
+  return jnp.empty_like(i).at[i].set(jnp.arange(i.size, dtype=i.dtype))
+
+@_wraps(scipy.stats.rankdata, lax_description="""\
+Currently the only supported nan_policy is 'propagate'
+""")
+@partial(jit, static_argnames=["method", "axis", "nan_policy"])
+def rankdata(
+  a: Array,
```

Comment:
We should generally use `ArrayLike` on function inputs (see https://jax.readthedocs.io/en/latest/jep/12049-type-annotations.html#inputs-should-be-permissively-typed)